pytorch_lightning/trainer/trainer.py

```diff
@@ -1171,15 +1174,15 @@ def _run(
             self._call_callback_hooks("on_fit_end")
             self._call_lightning_module_hook("on_fit_end")
 
-        # teardown if necessary (similar calls for spawn plugins are excluded as they have
-        # been included at the end of `new_process` functions)
-        if not isinstance(self.training_type_plugin, DDPSpawnPlugin):
-            self._call_teardown_hook()
+        self._call_teardown_hook()
 
         if self.state.status != TrainerStatus.INTERRUPTED:
             self.state.status = TrainerStatus.FINISHED
         self.state.stage = None
 
+        if isinstance(self.training_type_plugin, DDPSpawnPlugin):
```

Comment:
Any chance we can prevent strategy leakage here ?

Reply:
Not at this moment, but some ideas here #10985  might help to do that in the future.